lib/KTable/index.vue

```diff
@@ -0,0 +1,484 @@
+            v-for="(header, index) in headers"
+            :key="index"
+            tabindex="0"
+            :aria-sort="sortable && header.dataType !== DATA_TYPE_OTHERS ? getAriaSort(index) : null"
+            :class="{
+              [$computedClass(coreOutlineFocus)]: true,
+              sortable: sortable && header.dataType !== DATA_TYPE_OTHERS,
+              'sticky-header': true,
+              'sticky-column': index === 0,
+            }"
+            :style="[getHeaderStyle(header),
+                     sortKey === index ? { color: $themeBrand.primary.v_1000 } : { color: $themePalette.grey.v_800 },
+                     { backgroundColor: $themePalette.white } ,
+                     focusedColIndex === index ? { backgroundColor: $themePalette.grey.v_50 } : {}]"
+            role="columnheader"
+            data-focus="true"
+            :aria-colindex="index + 1"
+            @click="sortable && header.dataType !== DATA_TYPE_OTHERS ? handleSort(index) : null"
+            @keydown="handleKeydown($event, -1, index)"
+          >
+            <!--@slot Scoped slot for customizing the content of each header cell. Provides a header object `header` and its column index `colIndex`.-->
+
+            <slot name="header" :header="header" :index="index">
+              {{ header.label }}
+            </slot>
+            <span v-if="sortable && header.dataType !== DATA_TYPE_OTHERS" class="sort-icon">
+              <span v-if="sortKey === index && sortOrder === SORT_ORDER_ASC"><KIcon icon="dropup" :color="sortKey === index ? $themeBrand.primary.v_1100 : $themePalette.grey.v_800 " /></span>
+              <span v-else-if="sortKey === index && sortOrder === SORT_ORDER_DESC"><KIcon icon="dropdown" :color="sortKey === index ? $themeBrand.primary.v_1100 : $themePalette.grey.v_800 " /></span>
+              <span v-else><KIcon icon="sortColumn" :color="$themePalette.grey.v_800" /></span>
+            </span>
+          </th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr
+          v-for="(row, rowIndex) in finalRows"
+          :key="rowIndex"
+          :style="hoveredRowIndex === rowIndex || focusedRowIndex === rowIndex ? { backgroundColor: $themePalette.grey.v_50 } : {}"
+          @mouseover="handleRowMouseOver(rowIndex)"
+          @mouseleave="handleRowMouseLeave"
+        >
+          <KTableGridItem
+            v-for="(col, colIndex) in row"
+            :key="colIndex"
+            :content="col"
+            :dataType="headers[colIndex].dataType"
+            :minWidth="headers[colIndex].minWidth"
+            :width="headers[colIndex].width"
+            :rowIndex="rowIndex"
+            :colIndex="colIndex"
+            :class="{
+              'sticky-column': colIndex === 0,
+            }"
+            :style="[colIndex === 0 ? { backgroundColor: $themePalette.white } : {},(hoveredRowIndex === rowIndex || focusedRowIndex === rowIndex) && colIndex === 0 ? { backgroundColor: $themePalette.grey.v_50 } : {}
```

Comment:
Idem. Ternaries in the template of the component are a little bit harder to read. As this is a little more complex logic we could even add some comments in the computed prop to explain the logic. This could probably be simplified with `if ... else if ... else` as I see both items in the array are setting a backgroundColor.

Reply:
Added the logic as a computed property, using if-else statements.